ruby/computer_science/project_hash_map.md

```diff
@@ -4,50 +4,50 @@ You already know the magic behind hash maps, now it's time to write your own imp
 
-  1. Use the following snippet whenever you access a bucket through an index. We want to raise an error if we try to access an out of bound index:
+  Use the following snippet whenever you access a bucket through an index. We want to raise an error if we try to access an out of bound index:
 
-      ```ruby
-      raise IndexError if index.negative? || index >= @buckets.length
-      ```
+```ruby
+  raise IndexError if index.negative? || index >= @buckets.length
+```
 
 ### Assignment
 
 <div class="lesson-content__panel" markdown="1">
 
   1. Start by creating a `HashMap` class. Proceed to create the following methods:
 
-  1. `#hash` takes a value and produces a hash code with it. We did implement a fairly good `hash` method in the previous lesson. You are free to use that, or if you wish, you can conduct your own research. Beware this is a deep deep rabbit hole.
+  1. `#hash(key)` takes a key and produces a hash code with it. We did implement a fairly good `hash` method in the previous lesson. You are free to use that, or if you wish, you can conduct your own research. Beware this is a deep deep rabbit hole.
 
       <div class="lesson-note lesson-note--tip" markdown="1">
         Hash maps could accommodate various data types for keys like numbers, strings, and even other hashes. But for this project, only handle keys of type strings.
       </div>
 
-  1. `#set` takes two arguments, the first is a key and the second is a value that is assigned to this key. If a key already exists, then the old value is overwritten.
+  1. `#set(key, value)` takes two arguments, the first is a key and the second is a value that is assigned to this key. If a key already exists, then the old value is overwritten.
 
      - Remember to grow your buckets size when it needs to, by calculating if your bucket has reached the `load factor`.
 
-  1. `#get` takes one argument as a key and returns the value that is assigned to this key. If key is not found, return `nil`.
+  1. `#get(key)` takes one argument as a key and returns the value that is assigned to this key. If key is not found, return `nil`.
 
-  1. `#key?` takes a key as an argument and returns `true` or `false` based on whether or not the key is in the hash map.
+  1. `#has(key)` takes a key as an argument and returns `true` or `false` based on whether or not the key is in the hash map.
 
-  1. `#remove` takes a key as an argument. If the given key is in the hash map, it should remove the entry with that key and return the deleted entry's value. If the key isn't in the hash map, it should return `nil`.
+  1. `#remove(key)` takes a key as an argument. If the given key is in the hash map, it should remove the entry with that key and return the deleted entry's value. If the key isn't in the hash map, it should return `nil`.
 
-  1. `#length` returns the number of stored keys in the hash map.
+  1. `#length()` returns the number of stored keys in the hash map.
```

Comment:
For methods that don't take any arguments, it is convention in the Ruby language to omit the parentheses. So we'll want to go through those methods and pull the parentheses out.

The other changes you have here are great.